tests/test_safemath.py

```diff
@@ -105,3 +108,86 @@ async def test_sub_le_overflow(safemath_mock):
     b = to_uint(56789)
 
     await assert_revert(safemath.test_sub_le(a, b).invoke())
+
+
+@pytest.mark.asyncio
+async def test_mul(safemath_mock):
+    safemath = safemath_mock
+
+    a = to_uint(1234)
+    b = to_uint(56789)
+    c = mul_uint(a, b)
+
+    execution_info = await safemath.test_mul(a, b).invoke()
+    assert execution_info.result == (c,)
+
+
+@pytest.mark.asyncio
+async def test_mul_zero(safemath_mock):
+    safemath = safemath_mock
+
+    a = to_uint(0)
+    b = to_uint(56789)
+    c = to_uint(0)
+
+    execution_info = await safemath.test_mul(a, b).invoke()
+    assert execution_info.result == (c,)
+
+    execution_info = await safemath.test_mul(b, a).invoke()
+    assert execution_info.result == (c,)
+
+
+@pytest.mark.asyncio
+async def test_mul_overflow(safemath_mock):
+    safemath = safemath_mock
+
+    a = MAX_UINT256
+    b = to_uint(2)
+
+    await assert_revert(safemath.test_mul(a, b).invoke())
```

Comment:
should we add error messages or leave it for #190?

Reply:
Error messages for addition and subtraction are integrated in #190. I included error messages for multiplication and division in this PR